### PR TITLE
reduce use of Map.Topleft and Map.Bottomright

### PR DESCRIPTION
**Pull request by clemty**
as requested in #8549

### CODE REVIEW

File: mods/cnc/maps/gdi01/gdi01.lua [hunk 1]
```diff
@@ -49,11 +49,10 @@ Reinforce = function(units)
...
 
 CheckForBase = function()
-	baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
-		return actor.Type == "fact" or actor.Type == "pyle" or actor.Type == "nuke"
-	end)
-
-	return #baseBuildings >= 3
+	local barracks = player.GetActorsByType("pyle")
+	if #barracks > 0 then
+		return true
```
Comment (abcdefg30) [inline mods/cnc/maps/gdi01/gdi01.lua]: `return #barracks > 0`

File: mods/cnc/maps/nod05/nod05.lua [hunk 2]
```diff
@@ -270,10 +270,10 @@ IdleHunt = function(unit)
 end
 
 CheckForSams = function(player)
-	local baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
-		return actor.Owner == Nod and actor.Type == 'sam' end)
-
-	return #baseBuildings >= 3
+	local sams = Nod.GetActorsByType("sam")
+	if #sams >= 3 then
+		return true
```
Comment (abcdefg30) [inline mods/cnc/maps/nod05/nod05.lua]: `return #sams >= 3`

File: mods/cnc/maps/gdi01/gdi01.lua (hunk 1 above)
Comment (GraionDilach) [inline mods/cnc/maps/gdi01/gdi01.lua]: Can't it just be `return player.GetActorsByType("pyle") > 0`?

Comment (abcdefg30) [inline mods/cnc/maps/gdi01/gdi01.lua]: With a `#` before `player` (`return #player.GetActorsByType("pyle") > 0`), but yes, that is actually better.

Comment (clemty) [inline mods/cnc/maps/gdi01/gdi01.lua]: I find the current code easier to read - I can change it, but is it worth it in terms of runtime efficiency? (that function is only called once per second, and the GetActorsByType call is probably the bottleneck there)

**Comment by abcdefg30** [comment]
I'd also like it more if we keep `NodBaseBuildings = { "hand", "fact", "nuke" } ` and use this CheckForBase method:
```
CheckForBase = function(player)
	local buildings = 0

	Utils.Do(NodBaseBuildings, function(name)
		if #player.GetActorsByType(name) > 0 then
			buildings = buildings + 1
		end
	end)

	return buildings == #NodBaseBuildings
end

**Comment by clemty** [comment]
updated

@abcdefg30 : thanks for the code! I updated gdi01 as well to use that

**Comment by clemty** [comment]
updated with whitespace fixes (thanks @GraionDilach )
removed one instance of map.topleft/map.bottomright from allies03a

**Comment by abcdefg30** [comment]
Looks good to me and works. :+1: / :white_check_mark: 
Also thanks for the cleanup in allies03a.

**Comment by atlimit8** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/554c9dec5ba65f162a21d4b1734b9ef2b15886e7...cc998c71ca4e1357e5fafe3ed71874181f8e9f8f)

**Pull request merged**